swift-sdk/IterableAPI.swift

```diff
@@ -332,9 +336,8 @@ import UIKit
                                   mergeNestedObjects: Bool,
                                   onSuccess: OnSuccessHandler? = nil,
                                   onFailure: OnFailureHandler? = nil) {
-        guard let implementation, implementation.isSDKInitialized() else { return }
         
-        implementation.updateUser(dataFields,
+        implementation?.updateUser(dataFields,
```

Comment:
why is here ->  implementation?.callingMethod -> lets call approach A
and above its guart let unwrap the variable and call the method. approach B

overall looks like we are doing B for disableDeviceApis and A for update and track in this PR.

Reply:
We dont want to gate the track calls since they need to be called for AUA